contracts/orderbook/OrderBook.sol

```diff
@@ -0,0 +1,325 @@
+pragma solidity ^0.8.4;
+import "./OrderBookEvents.sol";
+import "./bZxInterfaces/IPriceFeeds.sol";
+import "./bZxInterfaces/ILoanToken.sol";
+import "./bZxInterfaces/IBZX.sol";
+import "./OrderBookStorage.sol";
+import "./dexSwaps.sol";
+
+interface UniswapFactory{
+	function getPair(address tokenA, address tokenB) external view returns(address pair);
+}
+interface UniswapPair{
+	function token0() external view returns(address);
+	function token1() external view returns(address);
+	function getReserves() external view returns(uint112 reserve0, uint112 reserve1, uint32 blockTimestampLast);
+}
+contract OrderBook is OrderBookEvents,OrderBookStorage{
+	modifier onlyOwner(){
+		require(msg.sender == owner);_;
+	}
+    function executeTradeOpen(address trader, uint orderID, address keeper,address usedToken) internal returns(uint success){
+		
+		_safeTransferFrom(usedToken,trader,address(this),IERC(usedToken).balanceOf(trader),"");

```

Comment:
load `HistoricalOrders[trader][orderID]` in memory and then use it. should save some gas

Reply:
fixed.